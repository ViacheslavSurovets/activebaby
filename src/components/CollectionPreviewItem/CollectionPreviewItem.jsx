import React from 'react';
import { connect } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { CloseIcon, CustomButton, ShopCartIcon } from '@components';
import {
  CollectionPreviewItemContainer,
  CollectionPreviewItemImage,
  CollectionPreviewItemTitle,
  CloseIconWrapper,
  CollectionPreviewItemHeader,
  CollectionPreviewItemImageWrapper,
  CustomButtonText,
  CollectionPreviewInfoText,
  CollectionPreviewInfoWrapper,
  CategoryItemTextWrapper,
  PreviewItemInfoWrapper,
  PreviewItemCustomButtonWrapper
} from './styles';
import {
  CategoryItemPriceText,
  CategoryItemText
} from '@components/CollectionOverviewItem/styles';
import PropTypes from 'prop-types';
import { createStructuredSelector } from 'reselect';
import { closePortalPreviewItem } from '@redux/previewItem/previewItem.actions';
import { selectPreviewItemData, selectPreviewItemInfo } from '@redux/previewItem/previewItem.selectors';


const CollectionPreviewItem = ( { item, info, closePortalPreviewItem } ) => {

  // const removeElem = async () => {
  //   let a = await document.getElementById ( 'portal' );
  //   a.parentNode.removeChild ( a );
  //   console.log ( a );
  // };

  const { t, i18n } = useTranslation ();
  return (
    <CollectionPreviewItemContainer>
      <CollectionPreviewItemHeader>
        <CollectionPreviewItemTitle>{ item.title }</CollectionPreviewItemTitle>
        <CloseIconWrapper onClick={ closePortalPreviewItem }>
          <CloseIcon color={ 'gray' } />
        </CloseIconWrapper>
      </CollectionPreviewItemHeader>

      <CollectionPreviewItemImageWrapper>
        <CollectionPreviewItemImage background={ item.images[ 0 ] } />
      </CollectionPreviewItemImageWrapper>

      <CollectionPreviewInfoWrapper>
        <CollectionPreviewInfoText>{ info }</CollectionPreviewInfoText>
      </CollectionPreviewInfoWrapper>

      <PreviewItemInfoWrapper>
        <CategoryItemTextWrapper>
          <CategoryItemText>
            <CategoryItemPriceText>{ t ( 'shopItem.price' ) }:</CategoryItemPriceText> { item.price } BYN
          </CategoryItemText>
        </CategoryItemTextWrapper>
        <PreviewItemCustomButtonWrapper>
          <CustomButton orangeSoft width={ '100%' } height={ '5rem' }>
            <CustomButtonText>{ t ( 'shopItem.toCart' ) }</CustomButtonText>
            <ShopCartIcon />
          </CustomButton>
        </PreviewItemCustomButtonWrapper>
      </PreviewItemInfoWrapper>
    </CollectionPreviewItemContainer>
  );
};


const mapStateToProps = createStructuredSelector({
  item: selectPreviewItemData,
  info: selectPreviewItemInfo
});


const mapDispatchToProps = dispatch => ({
  closePortalPreviewItem: () => dispatch ( closePortalPreviewItem () )
});

export default connect ( mapStateToProps, mapDispatchToProps ) ( CollectionPreviewItem );

CollectionPreviewItem.propTypes = {
  item: PropTypes.object,
  info: PropTypes.string,
  closePortalPreviewItem: PropTypes.func
};
