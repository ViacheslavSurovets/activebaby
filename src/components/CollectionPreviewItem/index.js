export { default as CollectionPreviewItem } from './CollectionPreviewItem';

