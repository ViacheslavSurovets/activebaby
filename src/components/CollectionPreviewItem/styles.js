import styled from 'styled-components';
import {
  commonStyleTextThreeRem,
  commonStyleTextTwoRem,
  flexCommon
} from '@src/styles/commonStylesAndStyledComponents';
import { tablet, mobile } from '@media';

export const CollectionPreviewItemContainer = styled.section`
  position: absolute;
  top: 50%;
  left:50%;
  margin-left:-27.5rem;
  margin-top: -26rem;
  display: flex;
  flex-direction: column;
  flex: 0 0 auto;
  background-color: ${ ( { theme } ) => theme.ghostWhite };
  z-index: 10;
  width: 55rem;
  ${ tablet }{
    width: 90%;
    margin-left: -46%
  };
  ${ mobile }{
    margin-left:-47%;
  }
`;

export const CollectionPreviewItemTitle = styled.span`
  ${ flexCommon }; 
  ${ commonStyleTextThreeRem };
`;


export const CollectionPreviewItemImage = styled.div`
  background-image: url(${ props => props.background });
  background-size: 100% 100%;
  width: 90%;
`;

export const CollectionPreviewItemHeader = styled.div`
  ${ flexCommon };
  position: relative;
`;

export const CloseIconWrapper = styled.div`
  align-self: flex-start;
  cursor: pointer;
  right:0;
  top:0;
  position: absolute;
`;

export const CollectionPreviewItemImageWrapper = styled.figure`
  display: flex;
  justify-content: center;
  width: 100%;
  height: 40rem;
  ${ mobile }{
    height: 80vw;
  }
`;

export const CustomButtonText = styled.span`
  ${ commonStyleTextTwoRem };
  margin-right: 2rem;
`;

export const CollectionPreviewInfoWrapper = styled.div`
  ${ flexCommon };
  width: 90%;
  margin: 1rem auto;
`;

export const CollectionPreviewInfoText = styled.span`
  ${ commonStyleTextTwoRem };
`;

export const CategoryItemTextWrapper = styled.div`
  margin-right:2rem;
`;

export const PreviewItemInfoWrapper = styled.div`
  display: flex;
  align-items: center; 
  justify-content: space-between;
  flex-wrap: wrap;
  ${ mobile }{
    justify-content: center;
  }
`;

export const PreviewItemCustomButtonWrapper = styled.div`
  width: 19rem;
  ${ mobile }{
    width: 100%;
    margin-top:.5rem;
  };
   &:hover {
   svg {
        fill: ${ ( { theme } ) => theme.header.cartColor };
      };
  }
`;
