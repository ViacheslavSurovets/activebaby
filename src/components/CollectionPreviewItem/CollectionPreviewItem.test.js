import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import CollectionPreviewItem from './CollectionPreviewItem';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('CollectionPreviewItem', ()=> {
  it('CollectionPreviewItem Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <CollectionPreviewItem  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
