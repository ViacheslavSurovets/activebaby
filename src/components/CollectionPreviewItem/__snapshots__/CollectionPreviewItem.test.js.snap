// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreviewItem CollectionPreviewItem Snapshot 1`] = `ShallowWrapper {}`;
